src/adapter/src/coord/command_handler.rs

```diff
@@ -134,11 +134,19 @@ impl Coordinator {
             }
 
             Command::SetSystemVars { vars, session, tx } => {
+                let config = self.catalog.system_config();
                 let ops = vars
                     .into_iter()
-                    .map(|(name, value)| catalog::Op::UpdateSystemConfiguration {
-                        name,
-                        value: OwnedVarInput::Flat(value),
+                    .map(|(name, value)| {
+                        let input = mz_sql::session::vars::VarInput::Flat(&value);
+                        if config.is_default(&name, input).unwrap_or(false) {
+                            catalog::Op::ResetSystemConfiguration { name }
+                        } else {
```

Comment:
I'm not sure I fully understand this, why do we use `ResetSystemConfiguration` instead of `UpdateSystemConfiguration`? They have slightly different semantics. `ResetSystemConfiguration` removes the variable from the stash while `UpdateSystemConfiguration` persists the variable with a value to the stash.

This also diverges from the behavior of  `sequence_alter_system_set` which doesn't check if the value is equal to the default value.

Reply:
Yeah, I think the current behavior here is probably better? Otherwise, if the compiled-in default changes in a new release of Materialize but _not_ LaunchDarkly, there will be a small moment during startup where the setting flips to the new default and then back to the LD override. Whereas with the current implementation, once LD sets the parameter to something, it sticks even if the binary compiles in a different default in a future versions.

Reply:
My point was that with the current implementation it seems that every feature flag value is going to live forever in the serialized stash. If we remove a value from the SystemVars fields, is it going to be somehow removed from its serialized version once we upgrade the catalog?

Reply:
Things removed from SystemVars will stay in the stash until explicitly removed. We can clean those up with migrations, or else with some code that deletes unknown keys when it's reading all of them.

Reply:
:point_up: @benesch given that the above change was not merged, maybe there should be an issue to track an alternative solution to the "growing stash" problem.